shortlister/tournament.py

```diff
@@ -0,0 +1,96 @@
+from itertools import combinations
+from pathlib import Path
+import pickle
+from typing import Dict, List
+from readchar import readkey
+
+
+def comparison(list_to_rank, result: Dict):
+    """Starts pair comparison for a list of items.
+    The result parameter could be dictionary with existing results or empty dictionary"""
+    # Queue is a list of pair that has not been compared yet
+    queue = get_pair(list_to_rank, result)
+
+    # Loop when there are still pairs to be compared in the queue
+    while len(queue) > 0:
+        # Remove the first pair in the queue and start the comparison
+        pair = tuple(queue.pop(0))
+        print(f"{pair[0]}[1] : {pair[1]}[2]")
+
+        # Limits key choice to "u","1","2" so there is no need for exception checking
+        while True:
+            choice = readkey()
+            if choice in ["u", "1", "2"]:
+                break
+        # if the choice is undo
+        if choice == "u" and result:
+            # take the key and requeue the pair
+            pair = result.popitem()[0]
+            queue.insert(0, pair)
+            continue
+        # otherwise it can be evaluate by choose() function to get the winner, and the result is saved
+        else:
+            winner = choose(choice, pair)
+            save_results(frozenset(pair), winner, result)
+
+
+def get_pair(list_to_rank, result):
+    """Return every unique item pair in the list."""
+    # Pairs are saved as frozenset here so that if order of list_to_rank is changed there won't be duplication pairs since "A","B" will be same as "B"","A"
+    unique_pairs = combinations(list_to_rank, 2)
+
```

Comment:
make all the pairs frozensets here, so you don't have to do it again in several places.